### PR TITLE
feat(checks): dump all checks as a json file

### PR DESCRIPTION
**Pull request by jchrisfarris**
### Context

This PR adds a CLI option to dump all the available checks into a json file that can be consumed by the `--checks-file` flag. 

### Description

```bash
prowler aws -jb > my_checks.json
# user edits my_checks.json to remove the ones they don't want
prowler aws --checks-file my_checks.json
```

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/__main__.py [hunk 1]
```diff
@@ -113,6 +113,12 @@ def prowler():
         provider,
     )
 
+    if args.list_checks_json:
```
Comment (jfagoagas) [inline prowler/__main__.py:118]: Could you include this within a function in the `check` lib?

I think you can reutilize the code there.

File: prowler/lib/cli/parser.py [hunk 2]
```diff
@@ -242,6 +242,9 @@ def __init_list_checks_parser__(self):
         list_group.add_argument(
             "-l", "--list-checks", action="store_true", help="List checks"
         )
+        list_group.add_argument(
```
Comment (jfagoagas) [inline prowler/lib/cli/parser.py:247]: For new flags and arguments we are not including short options for the flags, but we can leave it if you find it useful.

We have to include a test for this new option.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Thanks @jchrisfarris, this is great but please check the comments.

File: prowler/lib/cli/parser.py (hunk 2 above)
Comment (jchrisfarris) [reply prowler/lib/cli/parser.py:247]: Short code isn't needed. We can remove. I'll look at the other request when I circle back to this.

**Review by jfagoagas — APPROVED** [review]
Awesome, thanks @jchrisfarris !

**Pull request merged**